pkg/localstorev2/internal/reserve/reserve.go

```diff
@@ -0,0 +1,278 @@
+// Copyright 2023 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package reserve
+
+import (
+	"context"
+	"errors"
+	"sync"
+
+	"github.com/ethersphere/bee/pkg/localstorev2/internal"
+	"github.com/ethersphere/bee/pkg/log"
+	storagev2 "github.com/ethersphere/bee/pkg/storagev2"
+	"github.com/ethersphere/bee/pkg/swarm"
+	"github.com/ethersphere/bee/pkg/topology"
+)
+
+// loggerName is the tree path name of the logger for this package.
+const loggerName = "reserve"
+
+type Reserve struct {
```

Comment:
Would be nice to have tests for the `Reserve` methods.

Reply:
these are extensively tested through the localstorev2/reserve_test.go

Reply:
We need to have separate tests for this file. The localstorev2/reserve tests should only test the interaction with transactions, but the internal code needs to be tested separately.

```
PASS
	github.com/ethersphere/bee/pkg/localstorev2/internal/reserve	coverage: 27.6% of statements
ok  	github.com/ethersphere/bee/pkg/localstorev2/internal/reserve	0.125s	coverage: 27.6% of statements
```

There is no coverage reported here as some other package is testing this package. I think this is fundamentally wrong. If we make changes in the internal package, we will have to see which test to modify in the main one. I think this is a recipe for disaster. A seemingly harmless piece code in the old localstore caused leaks for years. If you dont want to do this in the current PR we can create a separate issue for this.

This is actually happening right now as there is a bug in the `EvictBatch` function as we are not deleting the batchRadiusItem. But this is not being checked anywhere.

Reply:
We are deleting it but ok